base/linalg/symmetric.jl

```diff
@@ -343,9 +343,24 @@ det(A::Symmetric) = det(bkfact(A))
 # ambiguity with RowVector
 \(A::HermOrSym{<:Any,<:StridedMatrix}, B::RowVector) = invoke(\, Tuple{AbstractMatrix, RowVector}, A, B)
 
-
-inv(A::Hermitian{T,S}) where {T<:BlasFloat,S<:StridedMatrix} = Hermitian{T,S}(inv(bkfact(A)), A.uplo)
-inv(A::Symmetric{T,S}) where {T<:BlasFloat,S<:StridedMatrix} = Symmetric{T,S}(inv(bkfact(A)), A.uplo)
+function _inv(A::HermOrSym)
+    n = checksquare(A)
+    B = inv!(lufact(A))
+    conjugate = isa(A, Hermitian)
+    # symmetrize
+    if A.uplo == 'U' # add to upper triangle
+        @inbounds for i = 1:(n-1), j = (i+1):n
+            B[i,j] = conjugate ? (B[i,j] + conj(B[j,i])) / 2 : (B[i,j] + B[j,i]) / 2
+        end
+    else # A.uplo == 'L', add to lower triangle
+        @inbounds for i = 1:(n-1), j = (i+1):n
+            B[j,i] = conjugate ? (B[j,i] + conj(B[i,j])) / 2 : (B[j,i] + B[i,j]) / 2
+        end
+    end
+    B
+end
+inv(A::Hermitian{T,S}) where {T<:BlasFloat,S<:StridedMatrix} = Hermitian{T,S}(_inv(A), A.uplo)
```

Comment:
This was here before this PR, but the construction of the `Hermitian` here elides the check for complex diagonal elements. Should that check be moved to an inner constructor? Or, alternatively, just call `Hermitian` here without type arguments, and rely on the outer constructor. Currently: 

```julia
julia> H
2×2 Hermitian{Complex{Float64},Array{Complex{Float64},2}}:
 0.650488-0.0im       0.826686+0.667447im
 0.826686-0.667447im   1.81707-0.0im     

julia> iH = inv(H)
2×2 Hermitian{Complex{Float64},Array{Complex{Float64},2}}:
  34.2282+2.13163e-14im  -15.5723-12.5727im    
 -15.5723+12.5727im       12.2532+7.10543e-15im

julia> isreal(iH[1,1])
false
```